app/src/main/java/com/example/henzoshimada/feeltrip/loginActivity.java

```diff
@@ -122,4 +316,88 @@ public void getUserInfo(ArrayList<Participant> participants){
             deleteRequestTask.execute(request);
         }
     }
+
+    public void swapToCustomTheme(View v) {
+        ColorPicker colorpicker = (ColorPicker) findViewById(R.id.picker);
+        SVBar sv = (SVBar) findViewById(R.id.svbar);
+        TableLayout themeSeekbarTable = (TableLayout) findViewById(R.id.theme_seekBar_table);
+        colorpicker.setVisibility(View.INVISIBLE);
+        sv.setVisibility(View.INVISIBLE);
+        themeSeekbarTable.setVisibility(View.VISIBLE);
+    }
+
+    public void swapToCustomThemeLight() {
+        setTheme(R.style.CustomTheme_Light);
+        FeelTripApplication.setThemeID(R.style.CustomTheme_Light);
+        Intent intent = new Intent(this, loginActivity.class);
+        Bundle bundle = new Bundle();
+        EditText userField = (EditText) this.findViewById(R.id.user_text);
+        EditText passField = (EditText) this.findViewById(R.id.pass_text);
+        bundle.putString("user",userField.getText().toString());
+        bundle.putString("pass",passField.getText().toString());
+        bundle.putString("custom","LIGHT");
+        intent.putExtras(bundle);
+        startActivity(intent);
+        finish();
+    }
```

Comment:
All of these are simply handlers for the application of custom themes, and background checking for the fade-in animation.